sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -378,6 +378,22 @@ class Analyzer(
         val newOrdering = resolveSortOrders(ordering, child, throws = false)
         Sort(newOrdering, global, child)
 
+      // A special case for Generate, because the output of Generate should not be resolved by
+      // ResolveReferences.
```

Comment:
This comment looks a little bit weird because this case is a part of ResolveReferences. Also, it will be good to say that `ResolveGenerate` will create `AttributeReference`s  for `output`. I will fix it when merging it.